compiler/resolution/implementForallIntents.cpp

```diff
@@ -909,7 +920,18 @@ static void checkAndRemoveOrigRetSym(Symbol* origRet, FnSymbol* parentFn) {
   std::vector<SymExpr*> symExprs;
   collectSymExprsSTL(parentFn, symExprs);
   for_vector(SymExpr, se, symExprs)
-    INT_ASSERT(se->var != origRet);
+    if (se->var == origRet) {
+      // It may appear in a no-init assignment.
+      bool OK = false;
+      if (CallExpr* parent = toCallExpr(se->parentExpr))
+        if (parent->isPrimitive(PRIM_MOVE))
+          if (CallExpr* rhs = toCallExpr(parent->get(2)))
+            if (rhs->isPrimitive(PRIM_NO_INIT)) {
```

Comment:
On the string-as-rec branch, we re-enabled PRIM_INIT initialization for RVVs of declared type, so this clause will probably have to be updated to include PRIM_INIT.  We want to remove that initialization anyway, and  this is a step in the right direction -- even if it applies only to yields.
Summary: Again, no action required: Just recording my thoughts.